File: src/main/java/no/nav/fo/veilarbregistrering/db/arbeidssoker/Formidlingsgruppeendring.java
package no.nav.fo.veilarbregistrering.db.arbeidssoker;

import java.sql.Timestamp;
import java.util.Comparator;
import java.util.Objects;

class Formidlingsgruppeendring {

    private final String formidlingsgruppe;
    private final int personId;
    private final String personIdStatus;
    private final Timestamp formidlingsgruppeEndret;

    public Formidlingsgruppeendring(String formidlingsgruppe, int personId, String personIdStatus, Timestamp formidlingsgruppeEndret) {
        this.formidlingsgruppe = formidlingsgruppe;
        this.personId = personId;
        this.personIdStatus = personIdStatus;
        this.formidlingsgruppeEndret = formidlingsgruppeEndret;
    }

    Timestamp getFormidlingsgruppeEndret() {
        return this.formidlingsgruppeEndret;
    }

    String getFormidlingsgruppe() {
        return this.formidlingsgruppe;
    }

    boolean erARBS() {
        return Objects.equals(formidlingsgruppe, "ARBS");
    }

    boolean erISERV() {
        return Objects.equals(formidlingsgruppe, "ISERV");
    }

    int getPersonId() {
        return personId;
    }

    String getPersonIdStatus() {
        return personIdStatus;
    }

    boolean erAktiv() {
        return Objects.equals(personIdStatus, "AKTIV");
    }

    @Override
    public String toString() {
        return "Formidlingsgruppeendring{" +
                "formidlingsgruppe='" + formidlingsgruppe + '\'' +
                ", personId=" + personId +
                ", personIdStatus='" + personIdStatus + '\'' +
                ", formidlingsgruppeEndret=" + formidlingsgruppeEndret +
                '}';
    }

    static class NyesteFoerst implements Comparator<Formidlingsgruppeendring> {

        static Formidlingsgruppeendring.NyesteFoerst nyesteFoerst() {
            return new Formidlingsgruppeendring.NyesteFoerst();
        }

        @Override
        public int compare(Formidlingsgruppeendring t0, Formidlingsgruppeendring t1) {
            return t1.getFormidlingsgruppeEndret().compareTo(t0.getFormidlingsgruppeEndret());
        }

    }
}

File: src/main/java/no/nav/fo/veilarbregistrering/db/arbeidssoker/Formidlingsgruppeendring.kt
package no.nav.fo.veilarbregistrering.db.arbeidssoker

import java.sql.Timestamp

internal class Formidlingsgruppeendring(
    val formidlingsgruppe: String,
    val personId: Int,
    val personIdStatus: String,
    val formidlingsgruppeEndret: Timestamp
) {
    fun erARBS(): Boolean {
        return formidlingsgruppe == "ARBS"
    }

    fun erISERV(): Boolean {
        return formidlingsgruppe == "ISERV"
    }

    fun erAktiv(): Boolean {
        return personIdStatus == "AKTIV"
    }

    override fun toString(): String {
        return "Formidlingsgruppeendring{" +
                "formidlingsgruppe='" + formidlingsgruppe + '\'' +
                ", personId=" + personId +
                ", personIdStatus='" + personIdStatus + '\'' +
                ", formidlingsgruppeEndret=" + formidlingsgruppeEndret +
                '}'
    }

    internal class NyesteFoerst : Comparator<Formidlingsgruppeendring> {
        override fun compare(t0: Formidlingsgruppeendring, t1: Formidlingsgruppeendring): Int {
            return t1.formidlingsgruppeEndret.compareTo(t0.formidlingsgruppeEndret)
        }

        companion object {
            fun nyesteFoerst(): NyesteFoerst {
                return NyesteFoerst()
            }
        }
    }
}